File: CSharp/Concentus/Celt/Structs/AnalysisInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Celt.Structs
{
    internal class AnalysisInfo
    {
        internal int valid = 0;
        internal float tonality = 0;
        internal float tonality_slope = 0;
        internal float noisiness = 0;
        internal float activity = 0;
        internal float music_prob = 0;
        internal int bandwidth = 0;

        internal AnalysisInfo()
        {
        }

        internal void Assign(AnalysisInfo other)
        {
            this.valid = other.valid;
            this.tonality = other.tonality;
            this.tonality_slope = other.tonality_slope;
            this.noisiness = other.noisiness;
            this.activity = other.activity;
            this.music_prob = other.music_prob;
            this.bandwidth = other.bandwidth;
        }

        internal void Reset()
        {
            valid = 0;
            tonality = 0;
            tonality_slope = 0;
            noisiness = 0;
            activity = 0;
            music_prob = 0;
            bandwidth = 0;
        }
    }
}

File: Go/opus/analysisInfo.go
package opus
type AnalysisInfo struct {
	enabled         bool
	valid           int
	tonality        float32
	tonality_slope  float32
	noisiness       float32
	activity        float32
	music_prob      float32
	bandwidth       int
}

func (a *AnalysisInfo) Assign(other *AnalysisInfo) {
	a.valid = other.valid
	a.tonality = other.tonality
	a.tonality_slope = other.tonality_slope
	a.noisiness = other.noisiness
	a.activity = other.activity
	a.music_prob = other.music_prob
	a.bandwidth = other.bandwidth
}

func (a *AnalysisInfo) Reset() {
	a.valid = 0
	a.tonality = 0
	a.tonality_slope = 0
	a.noisiness = 0
	a.activity = 0
	a.music_prob = 0
	a.bandwidth = 0
}